### ISSUE TITLE
Can't search for Phil Yu!

### ISSUE THREAD

**Issue by jgouly** [thewca/worldcubeassociation.org#2234]
I'm sure there are other names that don't search well, but this is the one I found.

https://www.worldcubeassociation.org/persons?page=1&region=all&search=phil+yu

Searching for the start of his ID doesn't work either:
https://www.worldcubeassociation.org/persons?page=1&region=all&search=yuphi

His profile is https://www.worldcubeassociation.org/persons/2010YUPH01

**Comment by jonatanklosko** [comment]
@jfly It probably has to do with #2196.

**Comment by larspetrus** [comment]
You _can_ search for the start of his ID: https://www.worldcubeassociation.org/persons?page=1&region=all&search=2010yup

**Comment by jgouly** [comment]
Good point, I actually meant 'part'. I edited my comment to change that.

**Comment by larspetrus** [comment]
I didn't mean to nitpick, just to inform :)

One of the changes in #2196 was that you can only search WCA IDs from the beginning, not for arbitrary substrings. This is a big performance win, but does have this usability loss.

I'll leave it to @jfly to investigate, but I'd speculate that the MYSQL full text search for names does something weird with 1-2 character strings.

**Comment by jfly** [comment]
Yeah, mysql's fulltext search defaults to a 4 letter minimum on words. I just checked our mysql configuration:

```
mysql> show variables like 'ft_min%';
+-----------------+-------+
| Variable_name   | Value |
+-----------------+-------+
| ft_min_word_len | 4     |
+-----------------+-------+
1 row in set (0.01 sec)
```

There's some information on changing this setting for AWS RDS here https://forums.aws.amazon.com/thread.jspa?threadID=92917 and here http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_WorkingWithParamGroups.html, but I'm not sure this is a rabbit hole we want to go down.

It's surprising to me that searching for a less than 4 letter word would fail, you'd expect it to just be ignored by mysql. The simplest possible thing we can do for now is to ignore search words less than 4 letters.

A later improvement might be to look into [AWS's managed elasticsearch solution](https://aws.amazon.com/elasticsearch-service/pricing/).

**Comment by larspetrus** [comment]
Lowering the MySQL limit is the better solution, but I'm not volunteering to figure out the RDS setting magic.

If you think that's a rabbit hole, Elastic Search is a rabbit Moria.

**Comment by larspetrus** [comment]
Because I only just realized it myself:

The "Phil Yu" search is failing because "Yu" simply isn't in the fulltext index. As far as the fulltext index is concerned his only name is "Phil".

**Comment by jgouly** [comment]
@jfly does the fulltext index always have to be used? Or can it be turned off for a particular query?

**Comment by jfly** [comment]
We introduced the fulltext index as a defense against some DOS attacks we were seeing a week ago, so turning off the fulltext index would open us up to those attacks again.

Someone just needs to take the time to dig into AWS RDS's settings.

**Comment by jgouly** [comment]
Ah, so turning it off for small searches wouldn't be a good idea. But for curiosity's sake, do you know if it's possible?

**Comment by jfly** [comment]
Yes, it's possible. As mentioned several times in this thread, take a look at #2196 if you want to see how we introduced the fulltext index.

**Comment by UnsolvedCypher** [comment]
I can't seem to reproduce the issue for either Yusheng Du or Phil Yu - is this still a problem?

**Comment by UnsolvedCypher** [comment]
@timreyn You're right- I didn't follow the above links and thought this was about search in the search bar, not the persons page search. Thanks!
